mypy/checker.py

```diff
@@ -1812,7 +1816,7 @@ def visit_decorator(self, e: Decorator) -> Type:
                 if d.fullname == 'typing.no_type_check':
                     e.var.type = AnyType()
                     e.var.is_ready = True
-                    return NoneTyp()
+                    return None
```

Comment:
See my comment above. The solution has to be a separate visitor, or at least it makes sense to me.